### PR TITLE
[WIP] Target down alert creation

### PR DESCRIPTION
**Pull request by obrienrobert**
## Description

- Creation of 2 new target down alerts: `TargetDown` and `BlackboxTargetDown`
- Update to the alerts exist test
- Each alert must be verified independently, with the steps to do so in the verification section

_**NOTE:**_ This PR should not be merged until a new release of the Keycloak Operator is created which includes the below mentioned Keycloak PR

## JIRA
[INTLY-9266](https://issues.redhat.com/browse/INTLY-9266)

Currently blocked by: [keycloak-operator/pull/243](https://github.com/keycloak/keycloak-operator/pull/243)

#### TargetDown
- Install using this branch, ensuring that the `TargetDown` alert isn't firing during install. It is OK if the `sso/usersso` pod monitor target alerts are firing - this is expected and the [Keyclock PR](https://github.com/keycloak/keycloak-operator/pull/243) removes these
- Get a list of all current Prometheus targets: `Prometheus -> Status -> Targets`
- To test the alert, change the metrics port number of one the `/metric` targets listed above. For example, to change the port number of  the `3scale-apicast-pods apicast staging` target, execute the below command and ensure that the alert is triggered:
```
oc patch dc apicast-staging -p '{"spec":{"template":{"metadata":{"annotations":{"prometheus.io/port":"9422"}}}}}' -n redhat-rhmi-3scale
```
- Once the alert has triggered, run the below command to reset the port number and ensure that the alert returns to normal:
```
oc patch dc apicast-staging -p '{"spec":{"template":{"metadata":{"annotations":{"prometheus.io/port":"9421"}}}}}' -n redhat-rhmi-3scale
```
-  Ensure the alert triggers and the alert message is clear, concise and descriptive. 

#### BlackboxTargetDown
- In order to get this alert to trigger, we need to change the port number that the Blackbox target uses (`9115 -> 9116`), which requires an update to the AMO as that is where port number gets set. For verification purposes, a pre-built image is available on Quay with this change
- In the `middleware-monitoring` namespace, navigate to `Installed Operators -> Application Monitoring Operator -> YAMK tab` and update the image used on line `11` and `268` to use the pre-built image: `quay.io/robrien/application-monitoring-operator:1.3.0`
- In Prometheus, ensure that the `BlackboxTarget` alert is firing for all targets (11 in total), and the alert is message is clear and concise

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: pkg/products/monitoring/prometheusRules.go [hunk 1]
```diff
@@ -282,6 +281,22 @@ func (r *Reconciler) newAlertsReconciler() resources.AlertReconciler {
...
 			},
+			{
+				AlertName: "general-monitoring-alerts",
+				Namespace: r.Config.GetOperatorNamespace(),
+				GroupName: "general.rules",
+				Rules: []monitoringv1.Rule{
+					{
+						Alert: "TargetDown",
+						Annotations: map[string]string{
+							"message": "{{ printf \"%.4g\" $value }}% of the {{ $labels.job }}/{{ $labels.service }} targets in {{ $labels.namespace }} namespace are down.",
```
Comment (Rajagopalan-Ranganathan) [inline pkg/products/monitoring/prometheusRules.go:292]: Remove the forward "/".
The namespace value is not displayed in the annotation of the alert.
Current message is shown as: 
```33.33% of the 3scale-apicast-pods/ targets in namespace are down.```

File: pkg/products/monitoring/prometheusRules.go [hunk 2]
```diff
@@ -282,6 +281,31 @@ func (r *Reconciler) newAlertsReconciler() resources.AlertReconciler {
...
+							"message": "{{ printf \"%.4g\" $value }}% of the {{ $labels.job }} targets in the {{ $labels.namespace }} namespace are down.",
+						},
+						Expr:   intstr.FromString("100 * (count(up == 0) BY (job, namespace, service) / count(up{job!=\"blackbox\"}) BY (job, namespace, service)) > 10"),
+						For:    "10m",
+						Labels: map[string]string{"severity": "warning"},
+					},
+					{
+						Alert: "BlackboxTargetDown",
+						Annotations: map[string]string{
+							"message": "The {{ $labels.service }} blackbox target is down.",
```
Comment (obrienrobert) [inline pkg/products/monitoring/prometheusRules.go:301]: @david-martin this is the placeholder `BlackboxTargetDown` message I've added. Any suggestions on something more descriptive?

Comment (david-martin) [reply pkg/products/monitoring/prometheusRules.go:301]: The url of the service would be good to show in the message (I believe that was one of the labels).

**Comment by jhellar** [comment]
Verified.
/lgtm

**Pull request closed without merging**